pytorch_lightning/plugins/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,22 @@
+from abc import ABC
+from pathlib import Path
+from typing import Any, Dict, Union
+
+
+class CheckpointIOPlugin(ABC):
+    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
```

Comment:
```suggestion
    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: Union[str, Path]) -> None:
```
`filepath` might assume the final destination is a file, which might not be true or might be abstracted away for other clients. any ideas for a name that could be more general than `filepath` ?

Reply:
maybe `output_location`?

Reply:
I wonder if `path` is a good candidate there. It is general enough but also less implies the local fs.